packages/payment/src/workflows/steps/create-payment-provider-session.ts

```diff
@@ -0,0 +1,39 @@
+import { PaymentProviderContext } from "@medusajs/types"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+interface StepInput {
+  provider_id: string
+  data: {
+    context: PaymentProviderContext
+    amount: number
+    currency_code: string
+  }
+}
+
+export const createPaymentProviderSessionStepId =
+  "module:create-payment-provider-session"
+export const createPaymentProviderSessionStep = createStep(
+  createPaymentProviderSessionStepId,
+  async (input: StepInput, { container }) => {
+    const service = container.resolve("paymentProviderService")
+
+    const session = await service.createSession(input.provider_id, input.data)
```

Comment:
here, as this is a workflow scoped to this module we could pass the sharedContext to call the service, and use the same transaction.
That could even allow us to remove the deleteSession on the compensate action, as that would be rollback by the same transaction.